Source/Particles/Collision/BinaryCollision/DSMC/SplitAndScatterFunc.H

```diff
@@ -0,0 +1,168 @@
+                p.atomicSetID(minus_one_long);
+            }
+
+            // Set the child particle properties appropriately
+            ptile1_data.m_rdata[PIdx::w][p_offsets[i] + ptile1_index] = p_pair_reaction_weight[i];
+            ptile2_data.m_rdata[PIdx::w][p_offsets[i] + ptile2_index] = p_pair_reaction_weight[i];
+
+            auto& ux1 = ptile1_data.m_rdata[PIdx::ux][p_offsets[i] + ptile1_index];
+            auto& uy1 = ptile1_data.m_rdata[PIdx::uy][p_offsets[i] + ptile1_index];
+            auto& uz1 = ptile1_data.m_rdata[PIdx::uz][p_offsets[i] + ptile1_index];
+            auto& ux2 = ptile2_data.m_rdata[PIdx::ux][p_offsets[i] + ptile2_index];
+            auto& uy2 = ptile2_data.m_rdata[PIdx::uy][p_offsets[i] + ptile2_index];
+            auto& uz2 = ptile2_data.m_rdata[PIdx::uz][p_offsets[i] + ptile2_index];
+
+            // for simplicity (for now) we assume non-relativistic particles
+            // and simply calculate the center-of-momentum velocity from the
+            // rest masses
+            auto const uCOM_x = (m1 * ux1 + m2 * ux2) / (m1 + m2);
+            auto const uCOM_y = (m1 * uy1 + m2 * uy2) / (m1 + m2);
+            auto const uCOM_z = (m1 * uz1 + m2 * uz2) / (m1 + m2);
+
+            // transform to COM frame
+            ux1 -= uCOM_x;
+            uy1 -= uCOM_y;
+            uz1 -= uCOM_z;
+            ux2 -= uCOM_x;
+            uy2 -= uCOM_y;
+            uz2 -= uCOM_z;
+
+            if (mask[i] == int(ScatteringProcessType::ELASTIC)) {
+                // randomly rotate the velocity vector for the first particle
+                ParticleUtils::RandomizeVelocity(
+                    ux1, uy1, uz1, std::sqrt(ux1*ux1 + uy1*uy1 + uz1*uz1), engine
+                );
+                // set the second particles velocity so that the total momentum
+                // is zero
+                ux2 = -ux1 * m1 / m2;
+                uy2 = -uy1 * m1 / m2;
+                uz2 = -uz1 * m1 / m2;
+            } else if (mask[i] == int(ScatteringProcessType::BACK)) {
+                // reverse the velocity vectors of both particles
+                ux1 *= -1.0_prt;
+                uy1 *= -1.0_prt;
+                uz1 *= -1.0_prt;
+                ux2 *= -1.0_prt;
+                uy2 *= -1.0_prt;
+                uz2 *= -1.0_prt;
+            } else if (mask[i] == int(ScatteringProcessType::CHARGE_EXCHANGE)) {
+                if (m1 == m2) {
+                    auto const temp_ux = ux1;
+                    auto const temp_uy = uy1;
+                    auto const temp_uz = uz1;
+                    ux1 = ux2;
+                    uy1 = uy2;
+                    uz1 = uz2;
+                    ux2 = temp_ux;
+                    uy2 = temp_uy;
+                    uz2 = temp_uz;
+                }
+                else {
+                    Abort("Uneven mass charge-exchange not implemented yet.");
+                }
+            }
+            else {
```

Comment:
It looks like `EXCITATION` was not implemented here. Is it intended?

Reply:
Yes, because in the DSMC scheme excitation would be a particle producing collision since the excited particle would need to be tracked as a separate kinetic species. With MCC since we don't track neutrals we can also ignore tracking excited neutrals and just assume they will relax before undergoing more collisions (since we are always "creating" a neutral to collide we just don't choose excited neutrals in that case).